tests/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2015, the Dartino project authors. Please see the AUTHORS file
+# for details. All rights reserved. Use of this source code is governed by a
+# BSD-style license that can be found in the LICENSE file.
+
+file(GLOB TESTS RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} "*-test.toit" ../third_party/sputnik/tests/sputnik_test*.toit)
+
+set(TOITRUN toit.run${CMAKE_EXECUTABLE_SUFFIX} CACHE FILEPATH "The executable used to run the tests")
+set(TOITPKG toit.pkg${CMAKE_EXECUTABLE_SUFFIX} CACHE FILEPATH "The executable used to install the packages")
+set(TEST_TIMEOUT 40 CACHE STRING "The maximal amount of time each test is allowed to run")
+set(SLOW_TEST_TIMEOUT 200 CACHE STRING "The maximal amount of time each slow test is allowed to run")
+
+message("TPKG: ${TOITPKG}")
+add_custom_target(
+  "install-pkgs"
+  COMMAND "${TOITPKG}" install
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+add_custom_target(
+  "install-pkgs-sputnik"
+  COMMAND "${TOITPKG}" install
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/sputnik/tests
+)
+
+include(ProcessorCount)
+ProcessorCount(NUM_CPU)
+
+add_custom_target(
+  check
+  COMMAND ${CMAKE_CTEST_COMMAND} -j${NUM_CPU} -T test --output-on-failure
```

Comment:
If you change to Makefile you also need to change this and give a default config `-C`.